### PR TITLE
Type: obj_factory for meta_vj

### PR DESCRIPTION
**Pull request by pbougue**

### CODE REVIEW

File: source/ed/build_helper.cpp [hunk 1]
```diff
@@ -126,25 +115,25 @@ VJ::VJ(builder & b, const std::string &line_name, const std::string &validity_pa
         name = "vehicle_journey " + std::to_string(vj->idx);
     }
 
-    nt::MetaVehicleJourney* mvj = b.get_or_create_metavj(name);
+    nt::MetaVehicleJourney* mvj = pt_data.meta_vj_fact.get_or_create(name);
```
Comment (antoine-de) [inline source/ed/build_helper.cpp]: rename `meta_vj_fact` to `meta_vjs`

File: source/ed/data.cpp [hunk 2]
```diff
@@ -578,8 +578,8 @@ void Data::build_associated_calendar() {
     size_t nb_matched(0);
     size_t nb_ignored(0);
 
-    for(auto meta_vj_pair : meta_vj_map) {
-        auto& meta_vj = meta_vj_map[meta_vj_pair.first]; //meta_vj_pair.second;
+    for(auto& meta_vj_pair : meta_vj_map) {
+        auto& meta_vj = meta_vj_pair.second;
```
Comment (antoine-de) [inline source/ed/data.cpp:582]: :+1:

File: source/ed/build_helper.cpp [hunk 3]
```diff
@@ -155,8 +144,8 @@ VJ::VJ(builder & b, const std::string &line_name, const std::string &validity_pa
         vj->set_vehicle(navitia::type::hasVehicleProperties::WHEELCHAIR_ACCESSIBLE);
     }
 
-    if(!b.data->pt_data->companies.empty())
-        vj->company = b.data->pt_data->companies.front();
+    if(!pt_data.companies.empty())
```
Comment (xlqian) [inline source/ed/build_helper.cpp]: spacing

File: source/ed/data.cpp (hunk 2 above)
Comment (xlqian) [inline source/ed/data.cpp:582]: :100:

File: source/type/pt_data.cpp [hunk 4]
```diff
@@ -33,8 +33,23 @@ www.navitia.io
...
+        if (vp->days == vp_ref.days && vp->beginning_date == vp_ref.beginning_date) {
+            return vp;
+        }
+    }
+    auto vp = new nt::ValidityPattern();
+    vp->idx = validity_patterns.size();
+    vp->uri = make_adapted_uri(vp->uri);
+    vp->beginning_date = vp_ref.beginning_date;
+    vp->days = vp_ref.days;
+    validity_patterns.push_back(vp);
```
Comment (xlqian) [inline source/type/pt_data.cpp:49]: i don't konw if you can use your ObjFactory for validity_patterns?

Comment (TeXitoi) [inline source/type/pt_data.cpp:49]: I don't think so, and that's out of the scope of this PR.

**Comment by TeXitoi** [comment]
OK for me.

File: source/type/pt_data.cpp (hunk 4 above)
Comment (xlqian) [inline source/type/pt_data.cpp:49]: ok

**Pull request merged**